src/server/scripts/Northrend/ChamberOfAspects/ObsidianSanctum/boss_sartharion.cpp

```diff
@@ -190,377 +205,415 @@ class boss_sartharion : public CreatureScript
+                    }
+                    // Handling of Drakes Events
+                    // Dragon Calls
+                    case EVENT_SARTHARION_CALL_TENEBRON:
+                    {
+                        Talk(SAY_SARTHARION_CALL_TENEBRON);
+                        if (Creature* tenebron = ObjectAccessor::GetCreature(*me, instance->GetData64(DATA_TENEBRON)))
+                        {
+                            tenebron->AI()->DoAction(ACTION_CALL_DRAGON);
+                        }
+                        break;
+                    }
+                    case EVENT_SARTHARION_CALL_SHADRON:
+                    {
+                        Talk(SAY_SARTHARION_CALL_SHADRON);
+                        if (Creature* shadron = ObjectAccessor::GetCreature(*me, instance->GetData64(DATA_SHADRON)))
+                        {
+                            shadron->AI()->DoAction(ACTION_CALL_DRAGON);
+                        }
+                        break;
+                    }
+                    case EVENT_SARTHARION_CALL_VESPERON:
+                    {
+                        Talk(SAY_SARTHARION_CALL_VESPERON);
+                        if (Creature* vesperon = ObjectAccessor::GetCreature(*me, instance->GetData64(DATA_VESPERON)))
+                        {
+                            vesperon->AI()->DoAction(ACTION_CALL_DRAGON);
+                        }
+
+                        break;
+                    }
+                }
             }
 
+            // Handle Sartharion combat abbilities
+            events.Update(diff);
+
             if (me->HasUnitState(UNIT_STATE_CASTING))
+            {
                 return;
+            }
 
-            HandleSartharionAbilities();
-            HandleDrakeAbilities();
+            while (uint32 const eventId = events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_SARTHARION_CAST_CLEAVE:
+                    {
+                        DoCastVictim(SPELL_SARTHARION_CLEAVE, false);
+                        events.RepeatEvent(10000);
+                    }break;
+                    case EVENT_SARTHARION_CAST_FLAME_BREATH:
+                    {
+                        DoCastVictim(SPELL_SARTHARION_FLAME_BREATH, false);
+                        events.RepeatEvent(20000);
+                    }break;
+                    case EVENT_SARTHARION_CAST_TAIL_LASH:
+                    {
+                        DoCastSelf(SPELL_SARTHARION_TAIL_LASH, false);
+                        events.RepeatEvent(18000);
+                    }break;
+                    case EVENT_SARTHARION_LAVA_STRIKE:
+                    {
+                        if (!urand(0, 2))
+                            Talk(SAY_SARTHARION_SPECIAL_4);
 
-            DoMeleeAttackIfReady();
-        }
-    };
-};
+                        summons.RemoveNotExisting();
+                        uint8 rand = urand(0, 4); // 5 - numer of cyclones
```

Comment:
we could use a constant `CYCLONES_COUNT`, have it set to 5 and then use `urand(0, CYCLONES_COUNT - 1)`